doc/Learn/Storage/HowTo-RequiredEntitlements.md

```diff
@@ -0,0 +1,64 @@
+---
+uid: Uno.Extensions.Storage.HowToRequiredEntitlements
+---
+# How-To: Adding Required Entitlements
+
+On Apple platforms (iOS, Mac Catalyst) the Uno storage extension use the OS Key Chain service to store secrets. This requires your application to have the [`keychain-access-groups`](https://developer.apple.com/documentation/bundleresources/entitlements/keychain-access-groups) entitlement to work properly.
+
+## Step-by-steps
+
+> [!IMPORTANT]
+> This guide assumes you used the template wizard or `dotnet new unoapp` to create your solution. If not, it is recommended that you follow the [instructions](xref:Uno.Extensions.HowToGettingStarted) for creating an application from the template.
+
+### 1. Adding the Entitlements.plist file
+
+The default location, inside your project, for the new file(s) should be:
+
+* iOS: `iOS/Entitlements.plist`
+* Mac Catalyst: `MacCatalyst/Entitlements.plist`
+
+The content of the file(s) should be:
+
+```xml
+<key>keychain-access-groups</key> 
+ <array> 
+ 	<string>$(AppIdentifierPrefix)$(CFBundleIdentifier)</string> 
+ </array>
+```
+
+For more information see Apple's documentation related to the [Key Chain](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps?language=objc).
+
+The variables `$(AppIdentifierPrefix)` and `$(CFBundleIdentifier)` will be replaced with the correct values at build time. For more information about how the Microsoft .NET SDK works with entitlements you can consult:
+
+* [Microsoft iOS](https://learn.microsoft.com/en-us/dotnet/maui/ios/entitlements)
+* [Microsoft Mac Catalyst](https://learn.microsoft.com/en-us/dotnet/maui/mac-catalyst/entitlements)
+
+### 2. Add capabilities in your Apple Developer Account
+
+Adding the `Entitlements.plist` to your project is not enough. You must also add the capability inside your Apple Developer Account and create a provisioning profile specific for your application. You can follow Microsoft's instruction for both steps:
+
+* [Microsoft iOS](https://learn.microsoft.com/en-us/dotnet/maui/ios/capabilities?#add-capabilities-in-your-apple-developer-account)
+* [Microsoft Mac Catalyst](https://learn.microsoft.com/en-us/dotnet/maui/mac-catalyst/capabilities?#add-capabilities-in-your-apple-developer-account)
+
```

Comment:
Should we be providing the XCode cheat way to get this to work? It seems that's easier than trying to follow the official documentation on how to do the manual steps

Reply:
Maybe? Many (dotnet) people have no experience with Xcode. It's also a workaround for development but to publish (TestFlight or App Store) the app you eventually have to do the "right thing". Having set it up properly earlier (and once) is less painful.

Still it can be a useful trick for other cases as well...

Reply:
added Xcode cheats